src/infrastructure/arranging/ordering.rs

```diff
@@ -432,49 +432,28 @@ mod tests {
         ( [$( $x:expr ),*] , [$( $y:expr),*] , $stream_id:expr) => {
         {
             // initialize vector of given range on the left.
-            let mut before: Vec<u16> = Vec::new();
-            $(
-                before.push($x);
-            )*
+            let before = [$($x,)*];
 
             // initialize vector of given range on the right.
-            let mut after: Vec<u16> = Vec::new();
-            $(
-                after.push($y);
-            )*
-
-            // generate test packets
-            let mut packets = Vec::new();
-            for (_, v) in before.iter().enumerate() {
-                packets.push(Packet::new(*v, $stream_id));
-            }
+            let after = [$($y,)*];
```

Comment:
Maybe I'm missing something but I can't see the place were after is filled with packets?

Reply:
`after` is not filled with packets, but with `u16`s.
so instead of
```rust
let mut after: Vec<u16> = Vec::new();
$(
  after.push($y);
)*
```
I simply wrote `let after = [$($y,)*];`

If your asking why I skip this:
```rust
let mut packets = Vec::new();
for (_, v) in before.iter().enumerate() {
  packets.push(Packet::new(*v, $stream_id));
}
```
I decided to construct packet in place, so instead of `stream.arrange(packet.sequence, packet.clone())` i simply construct it directly `stream.arrange(*seq, Packet::new(*seq, $stream_id))`

Reply:
Oke thanks for the comments